src/Operation/ChangeStream.php

```diff
@@ -0,0 +1,159 @@
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     */
+    public function __construct($databaseName, $collectionName, array $pipeline, array $options = [])
+    {
+        $expectedIndex = 0;
+
+        foreach ($pipeline as $i => $operation) {
+            if ($i !== $expectedIndex) {
+                throw new InvalidArgumentException(sprintf('$pipeline is not a list (unexpected index: "%s")', $i));
+            }
+
+            if ( ! is_array($operation) && ! is_object($operation)) {
+                throw InvalidArgumentException::invalidType(sprintf('$pipeline[%d]', $i), $operation, 'array or object');
+            }
+
+            $expectedIndex += 1;
+        }
+
+        if (isset($options['fullDocument']) && ! is_string($options['fullDocument'])) {
+            throw InvalidArgumentException::invalidType('"fullDocument" option', $options['fullDocument'], 'string');
+        }
+
+        if (isset($options['resumeAfter']) && ! is_array($options['resumeAfter']) && ! is_object($options['resumeAfter'])) {
+            throw InvalidArgumentException::invalidType('"resumeAfter" option', $options['resumeAfter'], 'array or object');
+        }
+
+        if (isset($options['maxAwaitTimeMS']) && ! is_integer($options['maxAwaitTimeMS'])) {
+            throw InvalidArgumentException::invalidType('"maxAwaitTimeMS" option', $options['maxAwaitTimeMS'], 'integer');
+        }
+
+        if (isset($options['batchSize']) && ! is_integer($options['batchSize'])) {
+            throw InvalidArgumentException::invalidType('"batchSize" option', $options['batchSize'], 'integer');
+        }
+
+        if (isset($options['collation']) && ! is_array($options['collation']) && ! is_object($options['collation'])) {
+            throw InvalidArgumentException::invalidType('"collation" option', $options['collation'], 'array or object');
+        }
+
+        $this->databaseName = (string) $databaseName;
+        $this->collectionName = (string) $collectionName;
+
+        $this->pipeline = $pipeline;
+
+        $this->options = $options;
+    }
+
+    /**
+     * Execute the operation.
+     *
+     * @see Executable::execute()
+     * @param Server $server
+     * @return Traversable
+     * @throws UnexpectedValueException if the command response was malformed
+     * @throws UnsupportedException if collation, read concern, or write concern is used and unsupported
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function execute(Server $server)
+    {
+        $command = $this->createCommand($server);
+
+        $cursor = $command->execute($server);
+
+        return $cursor;
+    }
+
+    /**
+     * Create the changeStream command.
+     *
+     * @param Server  $server
+     * @param boolean $isCursorSupported
+     * @return Command
+     */
+    private function createCommand(Server $server)
+    {
+        $changeStreamArray = ['$changeStream' => $this->options];
```

Comment:
Rather than pass all options into the pipeline operator, we should construct it with only known options that are `isset()`, as we do with command documents in other Operation classes.

Additionally, the options will need to be split based on which apply to the pipeline operator and which belong to Aggregate. `maxAwaitTimeMS` will likely need to be added to Aggregate down the line, since that's the Operation class that actually builds the Command object and will utilize the new API from [PHPC-1029](https://jira.mongodb.org/browse/PHPC-1029).